packages/remix/src/plugins/plugin.ts

```diff
@@ -0,0 +1,212 @@
+  buildTargetName?: string;
+  serveTargetName?: string;
+  startTargetName?: string;
+  typecheckTargetName?: string;
+}
+
+export const createNodes: CreateNodes<RemixPluginOptions> = [
+  '**/remix.config.{js,cjs}',
+  async (configFilePath, options, context) => {
+    const projectRoot = dirname(configFilePath);
+    const fullyQualifiedProjectRoot = join(context.workspaceRoot, projectRoot);
+    // Do not create a project if package.json and project.json isn't there
+    const siblingFiles = readdirSync(fullyQualifiedProjectRoot);
+    if (
+      !siblingFiles.includes('package.json') &&
+      !siblingFiles.includes('project.json')
+    ) {
+      return {};
+    }
+
+    options = normalizeOptions(options);
+
+    const hash = calculateHashForCreateNodes(projectRoot, options, context, [
+      getLockFileName(detectPackageManager(context.workspaceRoot)),
+    ]);
+    const targets = targetsCache[hash]
+      ? targetsCache[hash]
+      : await buildRemixTargets(configFilePath, projectRoot, options, context);
+
+    calculatedTargets[hash] = targets;
+
+    return {
+      projects: {
+        [projectRoot]: {
+          root: projectRoot,
+          targets,
+        },
+      },
+    };
+  },
+];
+
+async function buildRemixTargets(
+  configFilePath: string,
+  projectRoot: string,
+  options: RemixPluginOptions,
+  context: CreateNodesContext
+) {
+  const namedInputs = getNamedInputs(projectRoot, context);
+  const serverBuildPath = await getServerBuildPath(
+    configFilePath,
+    context.workspaceRoot
+  );
+
+  const targets: Record<string, TargetConfiguration> = {};
+  targets[options.buildTargetName] = buildTarget(
+    options.buildTargetName,
+    namedInputs
+  );
+  targets[options.serveTargetName] = serveTarget(serverBuildPath);
+  targets[options.startTargetName] = startTarget(
+    projectRoot,
+    serverBuildPath,
+    options.buildTargetName
+  );
+  targets[options.typecheckTargetName] = typecheckTarget(
+    projectRoot,
+    namedInputs
+  );
+
+  return targets;
+}
+
+function buildTarget(
+  buildTargetName: string,
+  namedInputs: { [inputName: string]: any[] }
+): TargetConfiguration {
+  return {
+    cache: true,
+    dependsOn: [`^${buildTargetName}`],
+    inputs: [
+      ...('production' in namedInputs
+        ? ['production', '^production']
+        : ['default', '^default']),
+    ],
+    outputs: ['{options.outputPath}'],
+    executor: '@nx/remix:build',
+    options: {
+      outputPath: '{workspaceRoot}/dist',
+    },
+  };
+}
+
+function serveTarget(serverBuildPath: string): TargetConfiguration {
+  return {
+    executor: '@nx/remix:serve',
+    options: {
+      command: `npx remix-serve ${serverBuildPath}`,
+      port: 4200,
+    },
+  };
+}
+
+function startTarget(
+  projectRoot: string,
+  serverBuildPath: string,
+  buildTargetName: string
+): TargetConfiguration {
+  return {
+    dependsOn: [buildTargetName],
+    command: `npx remix-serve ${serverBuildPath}`,
```

Comment:
do you explicitly use `npx`?

Reply:
Serve has this command passed to remix’s dev option which uses it under the hood to decide how to run the dev server.

start runs a production-ready server.